registration-processor/post-processor/registration-processor-printing-stage/src/main/java/io/mosip/registration/processor/print/stage/PrintingStage.java

```diff
@@ -233,66 +237,71 @@ public MessageDTO process(MessageDTO object) {
 				registrationStatusDto
 						.setLatestTransactionTypeCode(RegistrationTransactionTypeCode.PRINT_SERVICE.toString());
 
-			}
-			else {
-			String vid = getVid(uin);
-			requestWrapper.setId(env.getProperty("mosip.registration.processor.credential.request.service.id"));
-			String issuers=	env.getProperty(ISSUERS);
-			DateTimeFormatter format = DateTimeFormatter.ofPattern(env.getProperty(DATETIME_PATTERN));
-			LocalDateTime localdatetime = LocalDateTime
-						.parse(DateUtils.getUTCCurrentDateTimeString(env.getProperty(DATETIME_PATTERN)), format);
-			requestWrapper.setRequesttime(localdatetime);
-			requestWrapper.setVersion("1.0");
-			for (String key : issuers.split(SEMICOLON)) {
-				CredentialRequestDto credentialRequestDto=null;
-					String[] parts = key.split(COLON, 3);
-					credentialRequestDto = getCredentialRequestDto(regId, parts[0], parts[1], parts[2]);
+			} else {
+				String vid = getVid(uin);
+				requestWrapper.setId(env.getProperty("mosip.registration.processor.credential.request.service.id"));
+				String issuers=	env.getProperty(ISSUERS);
+				DateTimeFormatter format = DateTimeFormatter.ofPattern(env.getProperty(DATETIME_PATTERN));
+				requestWrapper.setVersion("1.0");
+				Set<String> credentialIssuerSet = new HashSet<>();
+				credentialIssuerSet.addAll(Arrays.asList(issuers.split(SEMICOLON)));
+				if (isCredentialShareEnabled) {
+					var metaInfo = utilities.getMetaInfo(regId, registrationStatusDto.getRegistrationType(), ProviderStageName.PRINTING);
```

Comment:
We need the identity json, but why we are talking the metainfo here ? Can we take this login into the getCredentialPartners method and use the below to pick fields actual field name as per the mapping file (we can also do this in the postconstruct)

https://github.com/mosip/registration/blob/develop/registration-processor/pre-processor/registration-processor-packet-classifier-stage/src/main/java/io/mosip/registration/processor/stages/packetclassifier/tagging/impl/IDObjectDataAvailabilityTagGenerator.java#L81-L84

And use the below get the values for those fiels, by specifically asking for it.

https://github.com/mosip/registration/blob/develop/registration-processor/pre-processor/registration-processor-packet-classifier-stage/src/main/java/io/mosip/registration/processor/stages/packetclassifier/PacketClassificationProcessor.java#L326

Reply:
Fetching MetaInfo in case "centerId" is the criteria to identify the credential partner.